masonry/src/widget/sized_box.rs

```diff
@@ -512,5 +554,131 @@ mod tests {
         assert_render_snapshot!(harness, "empty_box_with_gradient_background");
     }
 
+    #[test]
+    fn label_box_with_padding_and_background() {
+        let widget = SizedBox::new(Label::new("hello"))
+            .width(40.0)
+            .height(40.0)
+            .background(Color::PLUM)
+            .border(Color::LIGHT_SKY_BLUE, 5.)
+            .padding(100.);
+
+        let mut harness = TestHarness::create(widget);
+
+        assert_debug_snapshot!(harness.root_widget());
+        assert_render_snapshot!(harness, "label_box_with_background_and_padding");
+    }
+
+    #[test]
+    fn label_box_with_padding_outside() {
+        let widget = SizedBox::new(
+            SizedBox::new(Label::new("hello"))
+                .width(40.0)
+                .height(40.0)
+                .background(Color::PLUM)
+                .border(Color::LIGHT_SKY_BLUE, 5.),
+        )
+        .padding(100.);
+
+        let mut harness = TestHarness::create(widget);
+
+        assert_debug_snapshot!(harness.root_widget());
+        assert_render_snapshot!(harness, "label_box_with_outer_padding");
+    }
+
     // TODO - add screenshot tests for different brush types
 }
+
+// --- MARK: PADDING STRUCT ---
+
+/// Padding specifies the spacing between the edges of the box and the child view.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Padding {
+    /// The amount of padding for the top edge.
+    pub top: f64,
+    /// The amount of padding for the trailing edge.
+    ///
+    /// For LTR contexts this is the right edge, for RTL it is the left edge.
+    pub trailing: f64,
+    /// The amount of padding for the bottom edge.
+    pub bottom: f64,
+    /// The amount of padding for the leading edge.
+    ///
+    /// For LTR contexts this is the left edge, for RTL it is the right edge.
+    pub leading: f64,
+}
+
+impl Padding {
+    /// Creates a new [Padding] object by specifying the amount of padding for each edge.
```

Comment:
```suggestion
    /// Creates a new `Padding` by specifying the amount of padding for each edge.
```

See e.g. https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.new - if you're on this page, you already know you're talking about `Padding`

The term "object" also isn't needed or particularly meaningful in Rust.

Reply:
Makes sense :)